From e0a1237fa76d1d319d81d7200ac47c52ac2d48c8 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Mon, 6 Nov 2023 04:02:41 +0200
Subject: [PATCH] Android.mk changes to adapt for GloDroid

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 Android.mk      | 78 +++++++++++++++++++++++++++++++++----------------
 test/Android.mk | 16 +++++-----
 2 files changed, 61 insertions(+), 33 deletions(-)

diff --git a/Android.mk b/Android.mk
index d6dbf0a..9b8da43 100644
--- a/Android.mk
+++ b/Android.mk
@@ -158,7 +158,9 @@ LOCAL_CFLAGS := \
         -std=$(CPP_VERSION) \
         -fexceptions \
         -Werror \
-        -Wno-format-security
+        -Wno-format-security \
+        -no-Wunneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
 
 # Required to build with the changes made to the Android ML framework specific to Android R
 ifeq ($(ANDROID_R),1)
@@ -213,10 +215,8 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
-LOCAL_WHOLE_STATIC_LIBRARIES := libarmnn
 
 LOCAL_SHARED_LIBRARIES := \
         libbase \
@@ -227,7 +227,8 @@ LOCAL_SHARED_LIBRARIES := \
         libutils \
         android.hardware.neuralnetworks@1.0 \
         android.hidl.allocator@1.0 \
-        android.hidl.memory@1.0
+        android.hidl.memory@1.0 \
+        libarm_compute libarmnn
 
 ifeq ($(P_OR_LATER),1)
 # Required to build the 1.0 version of the NN Driver on Android P and later versions,
@@ -256,6 +257,8 @@ LOCAL_SHARED_LIBRARIES+= \
         libOpenCL
 endif
 
+LOCAL_MULTILIB := first
+
 include $(BUILD_STATIC_LIBRARY)
 
 ifeq ($(P_OR_LATER),1)
@@ -298,7 +301,9 @@ LOCAL_CFLAGS := \
         -Wextra \
         -Wno-unused-function \
         -Wno-format-security \
-        -DARMNN_ANDROID_NN_V1_1
+        -DARMNN_ANDROID_NN_V1_1 \
+        -Wno-unneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
 
 ifeq ($(ARMNN_DRIVER_DEBUG),1)
 LOCAL_CFLAGS+= \
@@ -355,11 +360,8 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
-LOCAL_WHOLE_STATIC_LIBRARIES := libarmnn
-
 LOCAL_SHARED_LIBRARIES := \
         libbase \
         libhidlbase \
@@ -370,7 +372,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.0 \
         android.hardware.neuralnetworks@1.1 \
         android.hidl.allocator@1.0 \
-        android.hidl.memory@1.0
+        android.hidl.memory@1.0 \
+        libarm_compute libarmnn
 
 ifeq ($(Q_OR_LATER),1)
 LOCAL_SHARED_LIBRARIES+= \
@@ -392,6 +395,8 @@ LOCAL_SHARED_LIBRARIES+= \
         libOpenCL
 endif
 
+LOCAL_MULTILIB := first
+
 include $(BUILD_STATIC_LIBRARY)
 
 endif # PLATFORM_VERSION == 9
@@ -430,7 +435,9 @@ LOCAL_CFLAGS := \
         -Wextra \
         -Wno-unused-function \
         -Wno-format-security \
-        -DARMNN_ANDROID_NN_V1_2
+        -DARMNN_ANDROID_NN_V1_2 \
+        -Wno-unneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
 
 ifeq ($(ARMNN_DRIVER_DEBUG),1)
 LOCAL_CFLAGS+= \
@@ -491,11 +498,8 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
-LOCAL_WHOLE_STATIC_LIBRARIES := libarmnn
-
 LOCAL_SHARED_LIBRARIES := \
         libbase \
         libhidlbase \
@@ -511,7 +515,9 @@ LOCAL_SHARED_LIBRARIES := \
         android.hidl.memory@1.0 \
         android.hardware.neuralnetworks@1.0 \
         android.hardware.neuralnetworks@1.1 \
-        android.hardware.neuralnetworks@1.2
+        android.hardware.neuralnetworks@1.2 \
+        libarm_compute libarmnn
+
 
 ifeq ($(R_OR_LATER),1)
 LOCAL_SHARED_LIBRARIES+= \
@@ -563,6 +569,9 @@ LOCAL_CFLAGS := \
         -Wno-unused-function \
         -Wno-format-security \
         -DARMNN_ANDROID_NN_V1_3 \
+        -Wno-unneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
+
 
 ifeq ($(ANDROID_R),1)
 LOCAL_CFLAGS+= \
@@ -629,11 +638,8 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
-LOCAL_WHOLE_STATIC_LIBRARIES := libarmnn
-
 LOCAL_SHARED_LIBRARIES := \
         libbase \
         libhidlbase \
@@ -651,13 +657,16 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.0 \
         android.hardware.neuralnetworks@1.1 \
         android.hardware.neuralnetworks@1.2 \
-        android.hardware.neuralnetworks@1.3
+        android.hardware.neuralnetworks@1.3 \
+        libarm_compute libarmnn
 
 ifeq ($(ARMNN_INCLUDE_LIBOPENCL),1)
 LOCAL_SHARED_LIBRARIES+= \
         libOpenCL
 endif
 
+LOCAL_MULTILIB := first
+
 include $(BUILD_STATIC_LIBRARY)
 
 endif # PLATFORM_VERSION == R
@@ -692,7 +701,9 @@ LOCAL_C_INCLUDES := \
 
 LOCAL_CFLAGS := \
         -std=$(CPP_VERSION) \
-        -fexceptions
+        -fexceptions \
+        -Wno-unneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
 
 ifeq ($(ARMNN_DRIVER_DEBUG),1)
 LOCAL_CFLAGS += \
@@ -718,7 +729,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -737,6 +747,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.0 \
         android.hidl.allocator@1.0 \
         android.hidl.memory@1.0 \
+        libarm_compute \
+        libarmnn \
         $(ARMNN_BACKEND_SHARED_LIBRARIES)
 
 ifeq ($(P_OR_LATER),1)
@@ -766,6 +778,8 @@ LOCAL_SHARED_LIBRARIES+= \
         libOpenCL
 endif
 
+LOCAL_MULTILIB := first
+
 include $(BUILD_EXECUTABLE)
 
 ifeq ($(P_OR_LATER),1)
@@ -802,7 +816,9 @@ LOCAL_C_INCLUDES := \
 LOCAL_CFLAGS := \
         -std=$(CPP_VERSION) \
         -fexceptions \
-        -DARMNN_ANDROID_NN_V1_1
+        -DARMNN_ANDROID_NN_V1_1 \
+        -Wno-unneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
 
 ifeq ($(ARMNN_DRIVER_DEBUG),1)
 LOCAL_CFLAGS += \
@@ -828,7 +844,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -848,6 +863,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.1 \
         android.hidl.allocator@1.0 \
         android.hidl.memory@1.0 \
+        libarm_compute \
+        libarmnn \
         $(ARMNN_BACKEND_SHARED_LIBRARIES)
 
 ifeq ($(Q_OR_LATER),1)
@@ -870,6 +887,8 @@ LOCAL_SHARED_LIBRARIES+= \
         libOpenCL
 endif
 
+LOCAL_MULTILIB := first
+
 include $(BUILD_EXECUTABLE)
 
 endif # PLATFORM_VERSION == 9
@@ -904,6 +923,8 @@ LOCAL_CFLAGS := \
         -std=$(CPP_VERSION) \
         -fexceptions \
         -DARMNN_ANDROID_NN_V1_2 \
+        -Wno-unneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
 
 ifeq ($(ARMNN_DRIVER_DEBUG),1)
 LOCAL_CFLAGS += \
@@ -929,7 +950,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -954,6 +974,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.0 \
         android.hardware.neuralnetworks@1.1 \
         android.hardware.neuralnetworks@1.2 \
+        libarm_compute \
+        libarmnn \
         $(ARMNN_BACKEND_SHARED_LIBRARIES)
 
 ifeq ($(R_OR_LATER),1)
@@ -967,6 +989,8 @@ LOCAL_SHARED_LIBRARIES+= \
         libOpenCL
 endif
 
+LOCAL_MULTILIB := first
+
 include $(BUILD_EXECUTABLE)
 
 endif # PLATFORM_VERSION == Q
@@ -1001,6 +1025,8 @@ LOCAL_CFLAGS := \
         -std=$(CPP_VERSION) \
         -fexceptions \
         -DARMNN_ANDROID_NN_V1_3 \
+        -Wno-unneeded-internal-declaration \
+        -Wno-unqualified-std-cast-call
 
 ifeq ($(ANDROID_R),1)
 LOCAL_CFLAGS+= \
@@ -1025,7 +1051,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -1052,6 +1077,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.1 \
         android.hardware.neuralnetworks@1.2 \
         android.hardware.neuralnetworks@1.3 \
+        libarm_compute \
+        libarmnn \
         $(ARMNN_BACKEND_SHARED_LIBRARIES)
 
 ifeq ($(ARMNN_INCLUDE_LIBOPENCL),1)
@@ -1059,6 +1086,8 @@ LOCAL_SHARED_LIBRARIES+= \
         libOpenCL
 endif
 
+LOCAL_MULTILIB := first
+
 include $(BUILD_EXECUTABLE)
 
 endif # PLATFORM_VERSION == R
@@ -1068,5 +1097,4 @@ endif # PLATFORM_VERSION == R
 ##########################
 # Note we use ANDROID_NN_DRIVER_LOCAL_PATH rather than LOCAL_PATH because LOCAL_PATH will be overwritten
 # when including other .mk files that set it.
-include $(ANDROID_NN_DRIVER_LOCAL_PATH)/armnn/Android.mk
 include $(ANDROID_NN_DRIVER_LOCAL_PATH)/test/Android.mk
diff --git a/test/Android.mk b/test/Android.mk
index 8621182..6f40a76 100644
--- a/test/Android.mk
+++ b/test/Android.mk
@@ -97,7 +97,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -113,7 +112,8 @@ LOCAL_SHARED_LIBRARIES := \
         libutils \
         android.hardware.neuralnetworks@1.0 \
         android.hidl.allocator@1.0 \
-        android.hidl.memory@1.0
+        android.hidl.memory@1.0 \
+        libarm_compute
 
 ifeq ($(P_OR_LATER),1)
 # Required to build the 1.0 version of the NN Driver on Android P and later versions,
@@ -230,7 +230,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -247,7 +246,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.0 \
         android.hardware.neuralnetworks@1.1 \
         android.hidl.allocator@1.0 \
-        android.hidl.memory@1.0
+        android.hidl.memory@1.0 \
+        libarm_compute
 
 ifeq ($(Q_OR_LATER),1)
 LOCAL_SHARED_LIBRARIES+= \
@@ -360,7 +360,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -382,7 +381,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.1 \
         android.hardware.neuralnetworks@1.2 \
         android.hidl.allocator@1.0 \
-        android.hidl.memory@1.0
+        android.hidl.memory@1.0 \
+        libarm_compute
 
 ifeq ($(R_OR_LATER),1)
 LOCAL_SHARED_LIBRARIES+= \
@@ -487,7 +487,6 @@ LOCAL_SRC_FILES := \
 LOCAL_STATIC_LIBRARIES := \
         libneuralnetworks_common \
         libflatbuffers-framework \
-        arm_compute_library \
         $(ARMNN_BACKEND_STATIC_LIBRARIES)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := \
@@ -511,7 +510,8 @@ LOCAL_SHARED_LIBRARIES := \
         android.hardware.neuralnetworks@1.2 \
         android.hardware.neuralnetworks@1.3 \
         android.hidl.allocator@1.0 \
-        android.hidl.memory@1.0
+        android.hidl.memory@1.0 \
+        libarm_compute
 
 ifeq ($(ARMNN_INCLUDE_LIBOPENCL),1)
 LOCAL_SHARED_LIBRARIES+= \
-- 
2.39.2

